### PR TITLE
impl. credential revocation with StatusList2021

### PR DESCRIPTION
**Pull request by MajdT51**

### CODE REVIEW

File: src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/services/RevocationServiceImpl.kt [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    override suspend fun addStatusEntry(profileName: String): CredentialStatus {
+        val httpResponse: HttpResponse = client.post {
+            url("$revocationUrl/management/lists/$profileName/entry")
+            accept(ContentType.Application.Json)
+            contentType(ContentType.Application.Json)
+        }
+        return Json.decodeFromString(httpResponse.readText())
+    }
+
+    override suspend fun getOwnStatusListCredential(listName: String): VerifiableCredentialDto {
```
[the remark this answers is not in the document]
Comment (MajdT51) [reply src/main/kotlin/org/eclipse/tractusx/managedidentitywallets/services/RevocationServiceImpl.kt:53]: renamed to getStatusListCredentialOfManagedWallet

File: src/test/kotlin/org/eclipse/tractusx/managedidentitywallets/PresentationsTest.kt [hunk 2]
```diff
@@ -874,4 +1034,514 @@ class PresentationsTest {
...
+                walletDto = EnvironmentTestSetup.walletService.createWallet(WalletCreateDto(EnvironmentTestSetup.DEFAULT_BPN, "base"))
+            }
+
+            SingletonTestData.baseWalletDID = walletDto.did
+            SingletonTestData.baseWalletVerKey = walletDto.verKey!!
+            SingletonTestData.credentialIndex = 3
+            SingletonTestData.revocationListName = walletDto.revocationListName!!
+
+            val verifiablePresentation = """
+                {
```
[the remark this answers is not in the document]
Comment (MajdT51) [reply src/test/kotlin/org/eclipse/tractusx/managedidentitywallets/PresentationsTest.kt:1065]: this will be done in a new branch

**Review by jpbu — COMMENTED** [review]
I have pushed some adjustments, and have an error in my local setup (see direct chat).

**Comment by jpbu** [comment]
I was able to successfully test it locally, a few minor comments left:
* We should use a proper version tag for the revocation service Docker image instead of "latest"
* When calling the get status list credential endpoint with an unknown list id, an error 500 is returned - should be 404
* We should double check which of the C-X issued credentials are revocable, I would argue that the BPN credential is not
* What is the reason to update the lists only once a day and not immediately after a revocation?

**Review by jpbu — APPROVED** [review]
Thanks, I reviewed the changes and verified locally, looks good now. Can be merged in coordination with the Helm / DevOps related changes.

**Pull request merged**